test/integration/etcd_storage_path_test.go

```diff
@@ -0,0 +1,1249 @@
+
+			if !ok {
+				t.Errorf("no test data for %s from %s.  Please add a test for your new type to etcdStorageData.", kind, pkgPath)
+				continue
+			}
+
+			if testData.ephemeral { // TODO it would be nice if we could remove this and infer if an object is not stored in etcd
+				t.Logf("Skipping test for %s from %s", kind, pkgPath)
+				continue
+			}
+
+			if isInKindAndPathWhiteList(kind, pkgPath) {
+				t.Logf("kind and path are whitelisted: skipping test for %s from %s", kind, pkgPath)
+				continue
+			}
+
+			if len(testData.expectedEtcdPath) == 0 || testData.stub == nil || reflect.TypeOf(testData.stub) != ptrType || isZero(reflect.ValueOf(testData.stub)) {
+				t.Errorf("invalid test data for %s from %s", kind, pkgPath)
+				continue
+			}
+
+			func() { // forces defer to run per iteration of the for loop
+				all := &[]runtime.Object{}
+				defer func() {
+					if !t.Failed() { // do not cleanup if test has already failed since we may need things in the etcd dump
+						if err := cleanup(f, testNamespace, all); err != nil {
+							t.Fatalf("failed to clean up etcd: %#v", err)
+						}
+					}
+				}()
+
+				if err := createPrerequisites(f, testNamespace, testData.prerequisites, all); err != nil {
+					t.Errorf("failed to create prerequisites for %s from %s: %#v", kind, pkgPath, err)
+					return
+				}
+
+				gvk := gv.WithKind(kind)
+
+				if err := create(f, testData.stub, &gvk, testNamespace, all); err != nil {
+					t.Errorf("failed to create stub for %s from %s: %#v", kind, pkgPath, err)
+					return
+				}
+
+				output, err := getFromEtcd(keys, d, testData.expectedEtcdPath, &gvk)
+				if err != nil {
+					t.Errorf("failed to get from etcd for %s from %s: %#v", kind, pkgPath, err)
+					return
+				}
+
+				// just check the type of whitelisted items
+				if isInCreateAndCompareWhiteList(output) {
+					outputType := reflect.TypeOf(output)
+					if outputType != ptrType {
+						t.Errorf("Output for %s from %s has the wrong type, expected %s, got %s", kind, pkgPath, ptrType.String(), outputType.String())
+					}
+					return
+				}
+
+				if !kapi.Semantic.DeepDerivative(testData.stub, output) {
+					t.Errorf("Test stub for %s from %s does not match: %s", kind, pkgPath, diff.ObjectGoPrintDiff(testData.stub, output))
+				}
+			}()
+		}
+	}
+
+	inEtcdData := diffMapKeys(etcdStorageData, seen)
+	inSeen := diffMapKeys(seen, etcdStorageData)
+	if len(inEtcdData) != 0 || len(inSeen) != 0 {
+		t.Fatalf("etcd data does not match the types we saw:\nin etcd data but not seen: %s\nseen but not in etcd data: %s", inEtcdData, inSeen)
+	}
+}
+
+func isInCreateAndCompareWhiteList(obj runtime.Object) bool {
+	switch obj.(type) {
+	case *authorizationapiv1.ClusterPolicyBinding, *authorizationapiv1.ClusterPolicy: // TODO figure out how to not whitelist these
+		return true
+	case *apisbatchv2alpha1.CronJob: // since we do not cleanup once a test is failed, we will get an AlreadyExists error since ScheduledJob aliases CronJob
```

Comment:
I stay strong, the less exceptions the better, currently imho there's too many. I don't see problems seeing multiple errors. Besides this will probably bite me, when dealing with those, or those dealing with it will ping me b/c job is failing ;)